### ISSUE TITLE
inotify wants str instead of bytes

### ISSUE THREAD

**Issue by skorpy2009** [mweinelt/kea-exporter#4]
> 15:04 < hexa- > hm, dann will das inotify package bei dir nun einen str haben, statt bytes

```
ffadmin@gw03:~$ /home/ffadmin/.local/bin/kea-exporter /etc/kea/kea-dhcp4.conf
Traceback (most recent call last):
  File "/home/ffadmin/.local/bin/kea-exporter", line 11, in <module>
    sys.exit(cli())
  File "/home/ffadmin/.local/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/ffadmin/.local/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/ffadmin/.local/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ffadmin/.local/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/ffadmin/.local/lib/python3.5/site-packages/kea_exporter/cli.py", line 17, in cli
    exporter = KeaExporter(config)
  File "/home/ffadmin/.local/lib/python3.5/site-packages/kea_exporter/kea.py", line 54, in __init__
    bytes(config_path, 'utf-8'), mask=inotify.constants.IN_MODIFY
  File "/home/ffadmin/.local/lib/python3.5/site-packages/inotify/adapters.py", line 93, in add_watch
    path_bytes = path_unicode.encode('utf8')
AttributeError: 'bytes' object has no attribute 'encode'
```

**Comment by mweinelt** [comment]
Except changing the parameter to `str` just gives a different exception:

```
Traceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/hexa/kea-exporter/kea_exporter/__main__.py", line 4, in <module>
    cli()
  File "/home/hexa/env/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/hexa/env/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/hexa/env/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/hexa/env/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/hexa/kea-exporter/kea_exporter/cli.py", line 17, in cli
    exporter = KeaExporter(config)
  File "/home/hexa/kea-exporter/kea_exporter/kea.py", line 54, in __init__
    config_path, mask=inotify.constants.IN_MODIFY
  File "/home/hexa/env/lib/python3.5/site-packages/inotify/adapters.py", line 67, in add_watch
    wd = inotify.calls.inotify_add_watch(self.__inotify_fd, path, mask)
ctypes.ArgumentError: argument 2: <class 'TypeError'>: wrong type
```

Can you post the output of `pip freeze`, please?

**Comment by skorpy2009** [comment]
```
ffadmin@gw03:~$ pip3 freeze
click==6.7
cryptography==1.7.1
hjson==3.0.1
idna==2.2
inotify==0.2.9
kea-exporter==0.3.0
keyring==10.1
keyrings.alt==1.3
netifaces==0.10.4
nose==1.3.7
prometheus-client==0.1.1
pyasn1==0.1.9
pycrypto==2.6.1
pygobject==3.22.0
python-apt==1.4.0b3
pyxdg==0.25
SecretStorage==2.3.1
six==1.10.0
unattended-upgrades==0.1
```

**Comment by mweinelt** [comment]
Ok, the reason this happens is because the `inotify` maintainer released `0.2.9` at the end of december 2017 and that changed the expected parameter type.

https://github.com/dsoprea/PyInotify/commit/b5735a7402f406675422f4bfa3d64c6f81814f4f

I'll push a fix any second now and probably do a `0.3.1` release later.
